crates/red_knot_python_semantic/resources/mdtest/import/star.md

```diff
@@ -549,26 +559,54 @@ from a import *
 
 reveal_type(X)  # revealed: bool
 
-# TODO: should emit error: [unresolved-reference]
+# error: [unresolved-reference]
 reveal_type(Y)  # revealed: Unknown
 
-# TODO: The `*` import should not be considered a redefinition
+# The `*` import should not be considered a redefinition
 # of the global variable in this module, as the symbol in
 # the `a` module is in a branch that is statically known
 # to be dead code given the `python-version` configuration.
-# Thus this should reveal `Literal[True]`.
-reveal_type(Z)  # revealed: Unknown
+# Thus this correctly reveals `Literal[True]` and does not have an error emitted
+reveal_type(Z)  # revealed: Literal[True]
 
 if sys.version_info >= (3, 12):
     from aa import *
 
-    # TODO: should reveal `Never`
+    # TODO: should not error because this branch is unreachable;
+    # should reveal `Never` (see https://github.com/astral-sh/ruff/issues/15967)
+    # error: [unresolved-reference]
     reveal_type(AA)  # revealed: Unknown
 
 # error: [unresolved-reference]
 reveal_type(AA)  # revealed: Unknown
 ```
 
+`c.py`:
+
+```py
+def coinflip() -> bool:
+    return True
+
+if coinflip():
+    Z: str = "Z"
+```
+
+`d.py`:
+
+```py
+Z = True
+
+from a import *
+from a import *
+from a import *
+
+reveal_type(Z)  # revealed: Literal[True]
+
+from c import *
+
+reveal_type(Z)  # revealed: Literal[True] | str
```

Comment:
Is this what you referred to in the PR description considering possibly-unbound imports?